tests/Silverback.Integration.Kafka.Tests/Messaging/Messages/KafkaKeyHelperTests.cs

```diff
@@ -58,5 +110,81 @@ public void GetMessageKey_MultipleKeyMembersMessagesWithSameKey_ComposedKeyIsRet
 
             key.Should().Be("One=1,Two=2");
         }
+
+        [Fact]
+        public void GetMessageKey_MultipleKeyMembersMessagesSecondEmpty_ComposedKeyIsReturned()
+        {
+            var message = new MultipleKeyMembersMessage
+            {
+                Id = Guid.NewGuid(),
+                One = string.Empty,
+                Two = "2",
+                Three = "3"
+            };
+
+            var key = KafkaKeyHelper.GetMessageKey(message);
+
+            key.Should().Be("Two=2");
+        }
+
+        [Fact]
+        public void GetMessageKey_DifferentMessagesMixture_CorrectKeyIsReturned()
+        {
+            // This is actually to test the cache.
+            var message1 = new MultipleKeyMembersMessage
+            {
+                Id = Guid.NewGuid(),
+                One = "1",
+                Two = "2",
+                Three = "3"
+            };
+
+            var message2 = new MultipleKeyMembersMessage
+            {
+                Id = Guid.NewGuid(),
+                One = "1",
+                Two = "2",
+            };
+
+            var message3 = new MultipleKeyMembersMessage
+            {
+                Id = Guid.NewGuid(),
+                One = "1",
+                Two = string.Empty,
+            };
+
+            var message4 = new MultipleKeyMembersMessage
+            {
+                Id = Guid.NewGuid(),
+                One = null,
+                Two = "2"
+            };
+
+            var key1 = KafkaKeyHelper.GetMessageKey(message1);
+            var key2 = KafkaKeyHelper.GetMessageKey(message2);
+            var key3 = KafkaKeyHelper.GetMessageKey(message3);
+            var key4 = KafkaKeyHelper.GetMessageKey(message4);
+
+            key1.Should().Be("One=1,Two=2");
+            key2.Should().Be("One=1,Two=2");
+            key3.Should().Be("One=1");
```

Comment:
Not sure if this is the desired behavior. Shouldn't it be like for a non-composite key if there is only one entry?

Reply:
I think it's correct., because the message actually has a key composed by the value of 2 properties and `{ One = "abc", Two = null }` should result in a different key than `{ One = null, Two = "abc" }`.

Reply:
On the other hand, wouldn't you expect a msg with only one key to end up with the same msg key as one with a composite key but one of the two props is null?

Reply:
Personally not, I think that in this case, if I annotate two properties, I would expect to be able to know where the value comes from.

I see there might be cases where you would want to have a sort of generic model with multiple properties annotated but only one set each time, depending maybe on the related entity type. But really, doesn't look to clean to me in any case and your point is valid only if in the same topic you write other messages with a different key configuration, which seems a bit of a mess to me.

There's no clear right and wrong in these cases, but I would leave it like this for backward compatibility in any case. Plus, if you want a clean key, you can work with a single property like the following example.

```csharp
class MyModel
{
   public string? A { get; set; }
   public string? B { get; set; }
   public string? C { get; set; }

   [KafkaKeyMember]
   [JsonIgnore]
   public string Key
   {
      get
      {
         if (...)
            return A;

         if (...)
            return B;

         ...
      }
   }
}
``` 

(Or you can use the enrichers via `WithKafkaKey` etc.)